pkg/streamingpromql/optimize/ast/testdata/reorder_histogram_aggregation.go

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package testdata
+
+var TestCasesReorderHistogramAggregation = map[string]string{
```

Comment:
There's two other edge cases we should consider:

* `histogram_sum(sum by (__name__) ({...}))` - we can't move the histogram function inside the aggregation as the function drops the metric name, which will cause incorrect aggregations
* `histogram_sum(sum({__name__=~"foo.*"}))` - we can't move the histogram function inside the aggregation as the function drops the metric name and the selector possibly selects metrics with different names, and the input to the aggregation could have multiple series with the same labels (which will cause an `vector cannot contain metrics with the same labelset` error)